ogc-common/src/main/java/io/arlas/server/ogc/common/requestfilter/FilterToElasticHelper.java

```diff
@@ -151,6 +159,17 @@ void visitGeoShapeBinarySpatialOperator(BinarySpatialOperator filter, Expression
             }
             e2.accept(delegate, extraData);
             shapeBuilder = delegate.currentShapeBuilder;
+            try {
+                GeoJsonObject object = GeoTypeMapper.getGeoJsonObject(shapeBuilder);
+                Geometry geometry = this.geometry.evaluate(object, Geometry.class);
+                CheckParams.checkWKT(geometry.toString());
+            } catch (ArlasException e) {
+                LOGGER.debug(e.getMessage());
+                List<OGCExceptionMessage> wfsExceptionMessages = new ArrayList<>();
+                wfsExceptionMessages.add(new OGCExceptionMessage(OGCExceptionCode.INVALID_PARAMETER_VALUE, e.getMessage() + " ["+ geometry + "]" , "filter"));
+                delegate.ogcException = new OGCException(wfsExceptionMessages, Service.WFS);
+                throw new RuntimeException();
```

Comment:
Why do you need to throw a RunTime Excp without text? It looks unusual

Reply:
FIxed